db/python/layers/project_insights.py

```diff
@@ -220,18 +234,25 @@ async def _stats_crams_query(
     AND a.type = 'CRAM'
     and a.status = 'COMPLETED';
         """
-        total_crams_by_project_id_and_seq_type = await self.connection.fetch_all(
+        _query_results = await self.connection.fetch_all(
             _query,
             {
-                'projects': projects,
-                'sequencing_types': sequencing_types,
+                "projects": project_ids,
+                "sequencing_types": sequencing_types,
             },
         )
-        return total_crams_by_project_id_and_seq_type
-
-    async def _stats_annotate_dataset_query(
-        self, projects: list[int], sequencing_types: list[str]
-    ):
+        # Return a dictionary with the project and sequencing fields as the key and a list of sequencing group ids as the value
+        crams_by_project_id_and_seq_fields = defaultdict(list)
+        for row in _query_results:
+            crams_by_project_id_and_seq_fields[
+                (row["project"], row["sequencing_type"], row["sequencing_technology"])
+            ].append(row["sequencing_group_id"])
+
+        return crams_by_project_id_and_seq_fields
+
+    async def _latest_annotate_dataset_by_project_id_and_seq_type(
+        self, project_ids: list[ProjectId], sequencing_types: list[str]
+    ) -> defaultdict[tuple[ProjectId, str], dict[str, Any]]:
```

Comment:
Because you use SequencingTechnology / Type so much, just declare an alias above to allow this to read easier